pkg/cli/zip_test.go

```diff
@@ -1252,3 +1252,60 @@ func TestCommandFlags(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// This tests the operation of zip over excluded nodes.
+func TestPartialZipForExcludedNodes(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	dir, cleanupFn := testutils.TempDir(t)
+	defer cleanupFn()
+	c := NewCLITest(TestCLIParams{
+		StoreSpecs: []base.StoreSpec{{
+			Path: dir,
+		}},
+	})
+	defer c.Cleanup()
+	zipName := filepath.Join(dir, "debug.zip")
+
+	// We want a low timeout so that the test doesn't take forever;
+	// however low timeouts make race runs flaky with false positives.
+	skip.UnderShort(t)
+	skip.UnderRace(t)
+
+	sc := log.ScopeWithoutShowLogs(t)
+	defer sc.Close(t)
+	// Reduce the number of output log files to just what's expected.
+	defer sc.SetupSingleFileLogging()()
+
+	ctx := context.Background()
+
+	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestIsSpecificToStorageLayerAndNeedsASystemTenant,
+			Insecure:          true,
+		},
+	})
+	defer tc.Stopper().Stop(ctx)
+
+	_, err := c.RunWithCapture("debug zip --concurrency=1 --exclude-nodes=1 --cpu-profile-duration=0 " + dir + "/debug.zip")
```

Comment:
We should probably test with a permutation of `--exclude-nodes` and `--nodes`

Reply:
We have range selection test:
https://github.com/cockroachdb/cockroach/blob/a44a9b1ffce25f51026b494a1dcb393cfc5361f3/pkg/cli/zip_test.go#L1060
One more challenge to club with current test is that `RunWithCapture` command runs against a single node even if we have a multi node cluster. So we are always getting end up with single node folder in debug zip.
Considering above both points, I didn't include in this test.
<!-- Reviewable comment -OB9h-9ZB4ctxblFsAu7:b-h7wa5j -->
<!-- Sent from Reviewable.io -->

Reply:
Ohh ok! Makes sense.
<!-- Reviewable comment -OB9nCB778a4fvBlcgQO:bwaxlak -->
<!-- Sent from Reviewable.io -->